packages/axelar-soroban-std/src/error.rs

```diff
@@ -52,23 +54,36 @@ macro_rules! assert_some {
     };
 }
 
-/// Assert that a [`Result`] is [`Err`] and matches a desired error
-#[macro_export]
-macro_rules! assert_err {
-    ( $x:expr, $e:expr ) => {
-        match $x {
-            std::result::Result::Err(e) => {
-                if $e != e {
-                    panic!("Expected error {}, got {:?} instead", stringify!($e), e)
-                }
-            }
-            std::result::Result::Ok(v) => {
-                panic!(
-                    "Expected error when calling {}, got {:?} instead",
-                    stringify!($x),
-                    v
-                );
+/// Assert that a [`Result`] is [`Err`] and matches a desired error.
+///
+/// `given` corresponds to the return type from `try_*` functions in Soroban.
+/// The function needs to fail, and successfully pass down the intended error type.
+/// So, for the assert to succeed, the parameters would be in the form:
+///
+/// given: `Err(Ok(ContractError))`
+/// expected: `ContractError`
+///
+/// Putting it together in a function call:
+///
+/// `assert_contract_err(client.try_fun(...), ContractError);`
+pub fn assert_contract_err<A, B, C, D>(given: Result<Result<A, B>, Result<C, D>>, expected: C)
```

Comment:
why not make it a macro? i'd prefer that for consistency

Reply:
I was thinking I wanted to enforce the type being passed into it and that a function would somewhat be self-documenting in that sense and be more helpful when used incorrectly. But I suppose it doesn't particularly matter either way, and the docstring should help with understanding usage. Rewrote it as a macro.